framework/rbac/DbManager.php

```diff
@@ -600,6 +608,10 @@ public function getAssignment($roleName, $userId)
         if (empty($userId)) {
             return null;
         }
+        if ($this->cacheAssignments) {
```

Comment:
If there's nothing in the cache? Nope. It will get to this query and execute it: https://github.com/yiisoft/yii2/pull/9138/files#diff-85a8556277e468a30afc79beb7aa25dfR681. This code which makes more optimal query https://github.com/yiisoft/yii2/pull/9138/files#diff-85a8556277e468a30afc79beb7aa25dfR653 would be never reached.